### PR TITLE
Distribcell Paths

### PR DESCRIPTION
**Pull request by wbinventor**
This PR removes the OpenCG dependency for distribcell paths in Tally Pandas DataFrames, as mentioned in #636.

### CODE REVIEW

File: docs/source/pythonapi/examples/pandas-dataframes.ipynb [hunk 1]
```diff
@@ -2089,29 +2090,29 @@
...
+       "568       0  10003   10001  16  12  0   10000  10002         284  absorption   \n",
+       "569       0  10003   10001  16  12  0   10000  10002         284     scatter   \n",
+       "570       0  10003   10001  16  13  0   10000  10002         285  absorption   \n",
+       "571       0  10003   10001  16  13  0   10000  10002         285     scatter   \n",
+       "572       0  10003   10001  16  14  0   10000  10002         286  absorption   \n",
+       "573       0  10003   10001  16  14  0   10000  10002         286     scatter   \n",
+       "574       0  10003   10001  16  15  0   10000  10002         287  absorption   \n",
+       "575       0  10003   10001  16  15  0   10000  10002         287     scatter   \n",
+       "576       0  10003   10001  16  16  0   10000  10002         288  absorption   \n",
+       "577       0  10003   10001  16  16  0   10000  10002         288     scatter   \n",
```
Comment (paulromano) [inline docs/source/pythonapi/examples/pandas-dataframes.ipynb]: Looks like y coords are flipped. Which one is correct?

File: docs/source/pythonapi/examples/pandas-dataframes.ipynb [hunk 2]
```diff
@@ -2271,7 +2272,7 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "Mann-Whitney Test p-value: 0.303583331507\n"
+      "Mann-Whitney Test p-value: 6.038663783e-42\n"
```
Comment (paulromano) [inline docs/source/pythonapi/examples/pandas-dataframes.ipynb]: This looks different too

File: docs/source/pythonapi/examples/pandas-dataframes.ipynb (hunk 1 above)
Comment (wbinventor) [inline docs/source/pythonapi/examples/pandas-dataframes.ipynb]: Oh yes, I meant to make a comment about this. The new convention is correct and reflects OpenMC's lattice cell indexing; the old scheme was actually incorrect as it used OpenCG's indexing.

File: docs/source/pythonapi/examples/pandas-dataframes.ipynb (hunk 2 above)
Comment (wbinventor) [inline docs/source/pythonapi/examples/pandas-dataframes.ipynb]: Excellent point, didn't realize I let this slip through the cracks. This diff is due to the comment I made above; I need to simply change flip the indexing in each code cell.

Comment (wbinventor) [inline docs/source/pythonapi/examples/pandas-dataframes.ipynb]: Fixed this with my latest commit.

File: openmc/filter.py [hunk 3]
```diff
@@ -531,12 +531,13 @@ def get_pandas_dataframe(self, data_size, summary=None):
         ----------
         data_size : Integral
             The total number of bins in the tally corresponding to this filter
-        summary : None or openmc.Summary
-            An optional Summary object to be used to construct columns for
-            distribcell tally filters (default is None). The geometric
+        distribcell_paths : bool
```
Comment (paulromano) [inline openmc/filter.py]: Put ", optional" at the end of this line and mention the default in the description. Also, is there a particular reason we shouldn't use the distribcell paths by default?

File: openmc/mgxs/mgxs.py [hunk 4]
```diff
@@ -1366,12 +1366,9 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
...
             Defaults to 'macro'.
-        summary : None or openmc.Summary
-            An optional Summary object to be used to construct columns for
-            distribcell tally filters (default is None). The geometric
-            information in the Summary object is embedded into a multi-index
-            column with a geometric "path" to each distribcell intance.
-            NOTE: This option requires the OpenCG Python package.
+    distribcell_paths : list of str
+        The paths traversed through the CSG tree to reach each distribcell
+        instance (for 'distribcell' filters only)
```
Comment (paulromano) [inline openmc/mgxs/mgxs.py]: Indentation is wrong on this block. Also, please indicate "optional" for arguments that have default values and list their defaults in the description.

File: openmc/mgxs/mgxs.py [hunk 5]
```diff
@@ -2577,12 +2577,9 @@ def get_pandas_dataframe(self, groups='all', nuclides='all',
...
```
Comment (paulromano) [inline openmc/mgxs/mgxs.py]: Same deal here w.r.t. indentation and optional

File: openmc/tallies.py [hunk 6]
```diff
@@ -1588,13 +1586,15 @@ def get_pandas_dataframe(self, filters=True, nuclides=True,
...
+        '''
         # If using Summary, ensure StatePoint.link_with_summary(...) was called
-        if summary and not self.with_summary:
+        if distribcell_pathssummary and not self.with_summary:
             msg = 'The Tally ID="{0}" has not been linked with the Summary. ' \
                   'Call the StatePoint.link_with_summary(...) method ' \
                   'before using Tally.get_pandas_dataframe(...) with ' \
                   'Summary info'.format(self.id)
             raise KeyError(msg)
+        '''
```
Comment (paulromano) [inline openmc/tallies.py]: Remove this block?

File: openmc/filter.py (hunk 3 above)
Comment (wbinventor) [inline openmc/filter.py]: I was thinking the same thing. Let's make the distribcell paths the default. Only reason I was leaving it as an option is due to the fact that building the multi-index for the paths isn't particularly efficient, and can be a bit slow for a large distribcell tally (*i.e.*, pick your favorite fuel pin in the full core BEAVRS model). However, the performance hit isn't noticeable for the smaller geometries that probably comprise the majority of use cases at this point. And I'm sure things can be sped up a great deal if this ever becomes a real pinch point in the future.

File: openmc/tallies.py (hunk 6 above)
Comment (wbinventor) [inline openmc/tallies.py]: Yes indeed

File: openmc/mgxs/mgxs.py (hunk 4 above)
Comment (wbinventor) [inline openmc/mgxs/mgxs.py]: Got it

**Comment by wbinventor** [comment]
@paulromano - this is up-to-date per your comments.

File: openmc/mgxs/mgxs.py (hunk 5 above)
Comment (paulromano) [inline openmc/mgxs/mgxs.py]: Indentation still looks wrong here

File: openmc/filter.py [hunk 7]
```diff
@@ -516,7 +516,7 @@ def get_bin(self, bin_index):
 
         return filter_bin
 
-    def get_pandas_dataframe(self, data_size, summary=None):
+    def get_pandas_dataframe(self, data_size, distribcell_paths=True):
```
Comment (paulromano) [inline openmc/filter.py]: The only problem with making this `True` by default is that if the `Summary` object is not linked it will fail.

Comment (wbinventor) [inline openmc/filter.py]: Agreed - I added in a check that the `Summary` info is linked.

File: openmc/mgxs/mgxs.py (hunk 5 above)
Comment (wbinventor) [inline openmc/mgxs/mgxs.py]: Should be fixed now.

File: openmc/filter.py [hunk 8]
```diff
@@ -626,106 +626,117 @@ def get_pandas_dataframe(self, data_size, summary=None):
...
-
-                # Initialize termination condition for while loop
-                levels_remain = True
-                counter = 0
+            # Create Pandas Multi-index columns for each level in CSG tree
+            if distribcell_paths:
+
+                # Distribcell paths require linked metadata from the Summary
+                if self.distribcell_paths is None:
+                    msg = 'Unable to construct distribcell paths since '
```
Comment (paulromano) [inline openmc/filter.py]: You need a forward slash at the end of this line or else `import openmc` will raise an IndentationError (in Python 3 at least).

Comment (wbinventor) [inline openmc/filter.py]: Yes, this is why Travis failed. This is what I get for relying too much on the pampering of PyCharm's IDE - I can't write a simple block of code in emacs anymore! This is fixed and hopefully the tests will pass.

**Comment by paulromano** [comment]
Looks like there's a conflict on the pandas dataframe notebook now since my PR changed it as well.

**Comment by wbinventor** [comment]
I fixed the conflict with pandas-dataframes.ipynb, but Travis isn't running now for some reason.

**Comment by paulromano** [comment]
The p-values in the notebook look very different again. Was there a change that got dropped when you merged with develop?

**Comment by wbinventor** [comment]
@paulromano - yes, I made a mistake when merging this with develop since there were some merge conflicts for the panda-dataframes notebook which were not simple to correct by hand. It should be up-to-date now.

**Comment by paulromano** [comment]
Thanks @wbinventor!

**Pull request merged**